test/MorphoActions.t.sol

```diff
@@ -0,0 +1,194 @@
+        assertApproxEqAbs(
+            IMorpho(morpho).position(marketId(marketParams), address(wallet)).borrowShares, 9.533e14, 0.1e14
+        );
+
+        vm.resumeGasMetering();
+        wallet.executeQuarkOperation(op, v, r, s);
+
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 200e6);
+        assertEq(IERC20(wstETH).balanceOf(address(wallet)), 5e18);
+    }
+
+    function testRepayAndWithdrawColalteralInShares() public {
+        vm.pauseGasMetering();
+        QuarkWallet wallet = QuarkWallet(factory.create(alice, address(0)));
+
+        deal(wstETH, address(wallet), 10e18);
+        deal(USDC, address(wallet), 100e6);
+        vm.startPrank(address(wallet));
+        IERC20(wstETH).approve(morpho, 10e18);
+        IMorpho(morpho).supplyCollateral(marketParams, 10e18, address(wallet), new bytes(0));
+        IMorpho(morpho).borrow(marketParams, 1000e6, 0, address(wallet), address(wallet));
+        vm.stopPrank();
+
+        uint256 sharesRepay = IMorpho(morpho).position(marketId(marketParams), address(wallet)).borrowShares;
+        // Repay all borrowed shares and withdraw all 10 wstETH collateral
+        QuarkWallet.QuarkOperation memory op = new QuarkOperationHelper().newBasicOpWithCalldata(
+            wallet,
+            MorphoActionsScripts,
+            abi.encodeWithSelector(
+                MorphoActions.repayAndWithdrawCollateral.selector, morpho, marketParams, 0, sharesRepay, 10e18
+            ),
+            ScriptType.ScriptSource
+        );
+        (uint8 v, bytes32 r, bytes32 s) = new SignatureHelper().signOp(alicePrivateKey, wallet, op);
+        assertEq(IERC20(USDC).balanceOf(address(wallet)), 1100e6);
+        assertEq(IERC20(wstETH).balanceOf(address(wallet)), 0);
+        assertApproxEqAbs(
+            IMorpho(morpho).position(marketId(marketParams), address(wallet)).borrowShares, 9.533e14, 0.1e14
+        );
+
+        vm.resumeGasMetering();
+        wallet.executeQuarkOperation(op, v, r, s);
+
+        assertApproxEqAbs(IERC20(USDC).balanceOf(address(wallet)), 100e6, 0.05e6);
+        assertEq(IMorpho(morpho).position(marketId(marketParams), address(wallet)).borrowShares, 0);
+        assertEq(IERC20(wstETH).balanceOf(address(wallet)), 10e18);
+    }
+
+    function testSupplyCollateralAndBorrow() public {
+        vm.pauseGasMetering();
+        QuarkWallet wallet = QuarkWallet(factory.create(alice, address(0)));
+
+        deal(wstETH, address(wallet), 10e18);
+
+        // Supply 10 wstETH as collateral and borrow 1000 USDC
+        QuarkWallet.QuarkOperation memory op = new QuarkOperationHelper().newBasicOpWithCalldata(
+            wallet,
+            MorphoActionsScripts,
+            abi.encodeWithSelector(
+                MorphoActions.supplyCollateralAndBorrow.selector, morpho, marketParams, 10e18, 1000e6
+            ),
+            ScriptType.ScriptSource
+        );
+        (uint8 v, bytes32 r, bytes32 s) = new SignatureHelper().signOp(alicePrivateKey, wallet, op);
```

Comment:
again, I find it confusing that we construct the op and it is immediately followed by assertions that are _pre_-conditions. I would prefer if we move this code down 3 lines and have the assertions of the "before" state come before the op.

Reply:
I sort of understand the reason behind this one that do all the generations including `op` beforehand. Then we group all the assertions at the end with one of the execution to split `before` and `after`.

For that I think let's do it in a separate PR as designated as `QoL PR: test refactor`
Because most of our current of the existing tests are doing in this order too,  so it would be better to just move everyone to the same ordering would be better imo. To be consistent across whole suites.

Reply:
I do not intend to kick off a large refactor across all our tests.

It's fine to drop this if we just don't want to make the change.

Reply:
Got it. Yeah usually I lean toward to keep consistent styles haha.  But I also open to create a refactor task to track (ofc it might not be in high priority but maybe worth picking it up sometime in the future)